packages/query/src/query/useList.ts

```diff
@@ -366,29 +367,52 @@ export const getFetchListQueryFn = (props: UseListProps) => {
       lte,
       not,
       select,
-      contains,
+      contains
     } = listFilters
 
+    const getSelectString = () => {
+      if (countOnly) {
+        return module?.select?.count || '*'
+      }
+
+      // @ts-ignore
+      const filterSelect = module?.select?.filter
+      if (filters?.length && filterSelect) {
+        return filterSelect
+      }
+
+      return select || module?.select?.list || '*'
+    }
+
```

Comment:
@clodal this allow `inner` to be selectedly added to the query when its in filter mode

Reply:
More info https://1xg.slack.com/archives/G01NU3G8H0Q/p1694577283747799